### PR TITLE
Remove iris.experimental.um module

### PR DESCRIPTION
**Pull request by dkillick**
Not strictly a deprecated module but it is part of `iris.experimental`, so may be changed or removed without warning. Removal of this clears part of the path for introducing `mule` for FF handling.

### CODE REVIEW

**Comment by pp-mo** [comment]
I think this is the wrong time for this, as it's not relevant to Iris 2.0.
It's presence is not an "obstacle" to making Iris use Mule, but that is now all post-2.0 anyway.
Regarding any direct user usage, Mule *is* already available to replace it, but we could helpfully first modify it to give a deprecation-type warning redirecting people to Mule, before actually removing it.

**Comment by pelson** [comment]
I can see both angles on this one. The idea of reducing the code-base of Iris is extremely appealing, but I do agree that removing this *now* for 2.0 doesn't buy us anything and adds *more* complexity to our 2.0 release notes.

@dkillick - happy to defer this?

**Comment by dkillick** [comment]
@pp-mo @pelson I'm sure you both asked me to look into removing this 😉 Happy to defer to v2.1 though.

**Comment by pelson** [comment]
Sorry if you've been getting mixed messages 😕 !
Very happy to merge this into master as soon as the v2.0.x branch is cut. Would you mind adding a what's new (under contributions_2.1)?

**Comment by cpelley** [comment]
Have we advertised that [experimental modules](http://scitools.org.uk/iris/docs/latest/iris/iris/experimental.html?highlight=experimental#module-iris.experimental) can disappear without notice?
I might have missed this, sorry if I have.

I would personally prefer deprecating a module regardless of whether in experimental or not.  Given the age of many of the experimental modules, many people are hooked into them (especially this one).  As a bonus it gives us the opportunity to point people to mule as part of a deprecation message rather than leaving them stranded.

**Comment by pelson** [comment]
> Have we advertised that experimental modules can disappear without notice?
I might have missed this, sorry if I have.

Yep: http://scitools.org.uk/iris/docs/latest/whitepapers/change_management.html#terminology-api-features-usages-and-behaviours

I'm 👎  on an enforced deprecation path for experimental (otherwise, what is the point of experimental at all?). We need to be able to change the experimental API at minor version changes, even if in practice we rarely do.

Having said all of that - I am happy enough to deprecate rather than remove where there is no significant burden on iris maintenance (always a hard thing to measure though)

**Comment by cpelley** [comment]
Thanks @pelson.  In that case I don't want to press further here.
I have opened a google groups iris-dev thread [here](https://groups.google.com/forum/#!topic/scitools-iris-dev/YTianJ7E3Gs) to continue this discussion independent to this PR.

Cheers

**Comment by dkillick** [comment]
@pelson whatsnew added 👍

File: docs/iris/src/whatsnew/contributions_2.1/incompatiblechange_2018-May-01_remove_experimental_um.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+* Removed :mod:`iris.experimental.um` to ease integration of Mule.
```
Comment (pp-mo) [inline docs/iris/src/whatsnew/contributions_2.1/incompatiblechange_2018-May-01_remove_experimental_um.txt]: Not sure about the "to ease..." bit.
I think this should instead explain in terms of what to replace with.
Perhaps? :
```"Removed :mod:`iris.experimental.um`.  Please use <https://github.com/SciTools/mule> instead."```

According to @stevewardle, that repo address is likely to be replaced soon by https://github.com/metomi/mule
 -- coming, but not ready yet.

Comment (dkillick) [reply docs/iris/src/whatsnew/contributions_2.1/incompatiblechange_2018-May-01_remove_experimental_um.txt]: @pp-mo yeah, I couldn't think of what to put here – it almost stopped at just "Removed :mod:`iris.experimental.um`". Your suggestion is good though, I'll put it in.

**Comment by pp-mo** [comment]
> Your suggestion is good though

Thanks for looking !
I'm sold ...

**Comment by pelson** [comment]
Excellent. This is my kind of PR. 🎉

**Comment by cpelley** [comment]
Sorry to jump in late here.  We are currently necessarily dependant on `iris.experimental.um` due to issues with the current mule version.  That is, we are waiting for the next release of mule (which we know has these issues fixed) to be able to migrate fully from `iris.experimental.um`.  Would you be able to re-instate this module until this release of mule is made available?

Cheers

**Comment by pelson** [comment]
> Would you be able to re-instate this module until this release of mule is made available?

I wouldn't be in favour of doing that. Obviously, ``iris.experimental.um`` is available in ``1.x`` and ``2.0.x``, and in an extreme situation, you could take the source of um and vendor it in your own tool.

Meanwhile, is there anything we can do to help mule get what you need?

**Comment by cpelley** [comment]
We would prefer not assume that we can pin iris if we can help it as we could find ourselves needing a newer release for a bugfix etc.
It's true that we can put this into our own code, but this isn't without cost.  We would need to expose this unfamiliar code to the public interface, which we would then need to deprecate in short order.
I assumed it would be easier (and make better sense) to delay the removal of this module (especially given its longevity) until this gap in functionality gets filled by the release of mule along with it becoming available externally.

> Meanwhile, is there anything we can do to help mule get what you need?

We have been in discussion with @bjlittle and @pp-mo around the availability of mule- thanks though.

**Comment by pelson** [comment]
I appreciate your very reasoned and level arguments, and want to emphasise that I'd only be in favour of reverting this PR in the last resort.

Essentially, I'd like to ensure the avenue of getting what you need into mule is entirely exhausted before bringing experimental code back into Iris, only for it to be removed again in the future.

@cpelley & @bjlittle - I'm depending on you both to make sure this moves forwards on the mule side of things. My support for reverting this PR would only come once I have clear evidence that the mule route has been entirely explored.

**Comment by cpelley** [comment]
Thanks for your engagement on this.
I only have an exposure to pieces of this problem, so this will be my last.
One of the points that I may not be making well is that getting mule such we don't require this module involves a great number of people, discussions and complexity (for example potentially changes to the UM I hear).

This is not something we can simply make happen right now or between myself and Bill.

I'm concerned that we should bare a cost to this decision right now.

Cheers

**Pull request merged**